library/spark/spark-library/src/main/java/uk/gov/gchq/gaffer/spark/operation/graphframe/GetGraphFrameOfElements.java

```diff
@@ -0,0 +1,152 @@
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.Lists;
+import org.apache.commons.collections.CollectionUtils;
+import org.graphframes.GraphFrame;
+
+import uk.gov.gchq.gaffer.commonutil.Required;
+import uk.gov.gchq.gaffer.data.element.id.DirectedType;
+import uk.gov.gchq.gaffer.data.elementdefinition.view.View;
+import uk.gov.gchq.gaffer.data.elementdefinition.view.ViewElementDefinition;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.graph.GraphFilters;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.spark.operation.dataframe.converter.property.Converter;
+import uk.gov.gchq.gaffer.spark.serialisation.TypeReferenceSparkImpl;
+import uk.gov.gchq.koryphe.ValidationResult;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An {@code Operation} that returns an Apache Spark {@code GraphFrame} (i.e. an
+ * abstraction over a {@link org.apache.spark.sql.Dataset} of {@link
+ * org.apache.spark.sql.Row}s) consisting of the {@code Elements}s converted to
+ * rows.
+ * <p>
+ * This GraphFrame object can be used as the basis for a number of graph
+ * processing queries.
+ *
+ * @see uk.gov.gchq.gaffer.spark.operation.dataframe.GetDataFrameOfElements
+ */
+public class GetGraphFrameOfElements implements
+        Output<GraphFrame>,
+        GraphFilters {
+
+    private List<Converter> converters;
+    private Map<String, String> options;
+    @Required
+    private View view;
+    private DirectedType directedType;
+
+    public GetGraphFrameOfElements() {
+    }
+
+    @Override
+    public TypeReference<GraphFrame> getOutputTypeReference() {
+        return new TypeReferenceSparkImpl.GraphFrame();
+    }
+
+    @Override
+    public Map<String, String> getOptions() {
+        return options;
+    }
+
+    @Override
+    public void setOptions(final Map<String, String> options) {
+        this.options = options;
+    }
+
+    public List<Converter> getConverters() {
+        return converters;
+    }
+
+    public void setConverters(final List<Converter> converters) {
+        this.converters = converters;
+    }
+
+    @Override
+    public View getView() {
+        return view;
+    }
+
+    @Override
+    public void setView(final View view) {
+        this.view = view;
+    }
+
+    @Override
+    public DirectedType getDirectedType() {
+        return directedType;
+    }
+
+    @Override
+    public void setDirectedType(final DirectedType directedType) {
+        this.directedType = directedType;
+    }
+
+    @Override
+    public ValidationResult validate() {
+        final ValidationResult result = Output.super.validate();
+
+        if (!view.hasEntities() || !view.hasEdges()) {
+            result.addError("Cannot create a Graphframe unless the View contains both edges and entities.");
+        }
+
+        final List<String> reservedProperties = Lists.newArrayList("id", "vertex", "group");
```

Comment:
See other comment. Could we change these to ID, VERTEX, GROUP?

Reply:
See my comment on your other comment.